src/union.ts

```diff
@@ -44,59 +44,52 @@ const createFSProxy = (watchers: FSWatcher[]) => new Proxy({}, {
     }
 });
 
-const fsPromisesMethods = [
-    'access',
-    'copyFile',
-    'open',
-    'opendir',
-    'rename',
-    'truncate',
-    'rmdir',
-    'mkdir',
-    'readdir',
-    'readlink',
-    'symlink',
-    'lstat',
-    'stat',
-    'link',
-    'unlink',
-    'chmod',
-    'lchmod',
-    'lchown',
-    'chown',
-    'utimes',
-    'realpath',
-    'mkdtemp',
-    'writeFile',
-    'appendFile',
-    'readFile'
-] as const;
+export type VolOptions = {
+    readonly?: boolean
```

Comment:
Since this is an object, whats the expected behaviour of this:

```
{
  readonly: true,
  writeonly: true,
}
```

I think it would be better to either use a string union:

```
type Behaviour: 'readonly' | 'writeonly'
```

or make these `writable` & `readable`:

```
type VolOptions {
  writable?: boolean;
  readable?: boolean;
}
```

Both are fine, but the `VolOptions` one would then let us just say "`writable` & `readable` are both `true` by default", leading to simplified behaviour elsewhere as we could just check for the appropriate option when needed.

Reply:
Have added a check in the .use() parameter that will throw a logic error. Test included